src/Files.App/Filesystem/StorageItems/NativeStorageFile.cs

```diff
@@ -143,18 +140,15 @@ public override IAsyncOperation<StorageItemThumbnail> GetThumbnailAsync(Thumbnai
 		public override IAsyncOperation<StorageItemThumbnail> GetThumbnailAsync(ThumbnailMode mode, uint requestedSize, ThumbnailOptions options)
 			=> throw new NotSupportedException();
 
-		public static IAsyncOperation<BaseStorageFile> FromPathAsync(string path)
+		public static IAsyncOperation<BaseStorageFile>? FromPathAsync(string path)
 		{
-			return AsyncInfo.Run<BaseStorageFile>(async (cancellationToken) =>
+			if (IsNativePath(path) && CheckAccess(path))
 			{
-				if (IsNativePath(path) && CheckAccess(path))
-				{
-					var name = IO.Path.GetFileName(path);
-					return new NativeStorageFile(path, name.Substring(name.LastIndexOf(":") + 1), DateTime.Now);
-				}
-				return null;
-			});
-		}
+				var name = IO.Path.GetFileName(path);
+				return Task.FromResult((BaseStorageFile)new NativeStorageFile(path, name[(name.LastIndexOf(":") + 1)..], DateTime.Now)).AsAsyncOperation();
+			}
+            return null;
```

Comment:
They are not equivalent, instead using

```suggestion
            return Task.FromResult<BaseStorageFile>(null).AsAsyncOperation();
```